tests/tests_data/streaming/test_dataset.py

```diff
@@ -127,17 +72,17 @@ def test_streaming_dataset_distributed_no_shuffle(tmpdir):
     assert len(dataset) == 101
 
     dataset.distributed_env = _DistributedEnv(2, 0)
```

Comment:
In this case, will `_DistributedEnv(2, 1)` get length 50? Since `drop_last=False`. Could you assert the correct value for rank 1?